salto-io/salto
# SALTO-4579 - Salesforce: Fetch instances based on a configurable field

Introduce the `saltoManagementFieldSettings.defaultFieldName` config. Fetching instances would now work in the following way:

1. If the instance's type is in `allowReferenceTo` and there's a reference to the instance, fetch it
2. If the instance's type is in `allowReferenceTo` and the instance has the `saltoManagementField` and the field's value is `true`, fetch it
3. If the instance's type is in `excludeObjects`, don't fetch it
4. If the instance's type is in `includeObjects`, and the instance has the `saltoManagementField` and the field's value is `true`, fetch it.
5. If the instance's type is in `includeObjects`, and does not have the `saltoManagementField`, fetch it.
6. Otherwise, don't fetch it.

---

~Note that this PR builds on top of https://github.com/salto-io/salto/pull/4762 - the first two commits (and any changes to `custom_objects_instances_references.ts`) are from the other PR.~

~Unit tests for `custom_objects_instances` are still pending, but I figured it's worth starting the review.~

Tests I ran include:
* Fetch without the config value and with the type in `includeObjects`, verify all instances are fetched
* Fetch with the config value and a type in `includeObjects`, verify only instances of the type where the management field is `true` are fetched.
* Fetch with the config value and a type in `allowReferencesTo`, verify only instances of the type where the management field is `true` are fetched even though there are no references to them.

---
_Release Notes_: 
Salesforce: Introduced the `saltoManagementFieldSettings` data configuration section. Use the `defaultFieldName` entry in this section to set the name of the "Managed By Salto" field. If this configuration entry is set, only records where this field is set to `true` will be fetched.

---
_User Notifications_: 
N/A
